Managers/PrimitiveMan.cpp

```diff
@@ -0,0 +1,230 @@
+		if (!g_SceneMan.SceneWrapsX() && !g_SceneMan.SceneWrapsY()) {
+			Vector drawStart = m_StartPos - targetPos;
+			circle(pDrawScreen, drawStart.m_X, drawStart.m_Y, m_Radius, m_Color);
+		} else {
+			Vector drawStartLeft;
+			Vector drawStartRight;
+
+			TranslateCoordinates(targetPos, m_StartPos, drawStartLeft, drawStartRight);
+
+			circle(pDrawScreen, drawStartLeft.m_X, drawStartLeft.m_Y, m_Radius, m_Color);
+			circle(pDrawScreen, drawStartRight.m_X, drawStartRight.m_Y, m_Radius, m_Color);
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void PrimitiveMan::CircleFillPrimitive::Draw(BITMAP *pDrawScreen, Vector targetPos) {
+		if (!g_SceneMan.SceneWrapsX() && !g_SceneMan.SceneWrapsY()) {
+			Vector drawStart = m_StartPos - targetPos;
+			circlefill(pDrawScreen, drawStart.m_X, drawStart.m_Y, m_Radius, m_Color);
+		} else {
+			Vector drawStartLeft;
+			Vector drawStartRight;
+
+			TranslateCoordinates(targetPos, m_StartPos, drawStartLeft, drawStartRight);
+
+			circlefill(pDrawScreen, drawStartLeft.m_X, drawStartLeft.m_Y, m_Radius, m_Color);
+			circlefill(pDrawScreen, drawStartRight.m_X, drawStartRight.m_Y, m_Radius, m_Color);
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void PrimitiveMan::TextPrimitive::Draw(BITMAP *pDrawScreen, Vector targetPos) {
+		if (!g_SceneMan.SceneWrapsX() && !g_SceneMan.SceneWrapsY()) {
+			Vector drawStart = m_StartPos - targetPos;
+			AllegroBitmap pPlayerGUIBitmap(pDrawScreen);
+
+			if (m_IsSmall) {
+				g_FrameMan.GetSmallFont()->DrawAligned(&pPlayerGUIBitmap, drawStart.m_X, drawStart.m_Y, m_Text, m_Alignment);
+			} else {
+				g_FrameMan.GetLargeFont()->DrawAligned(&pPlayerGUIBitmap, drawStart.m_X, drawStart.m_Y, m_Text, m_Alignment);
+			}
+		} else {
+			Vector drawStartLeft;
+			Vector drawStartRight;
+
+			TranslateCoordinates(targetPos, m_StartPos, drawStartLeft, drawStartRight);
+
+			AllegroBitmap pPlayerGUIBitmap(pDrawScreen);
+			if (m_IsSmall) {
+				g_FrameMan.GetSmallFont()->DrawAligned(&pPlayerGUIBitmap, drawStartLeft.m_X, drawStartLeft.m_Y, m_Text, m_Alignment);
+				g_FrameMan.GetSmallFont()->DrawAligned(&pPlayerGUIBitmap, drawStartRight.m_X, drawStartRight.m_Y, m_Text, m_Alignment);
+			} else {
+				g_FrameMan.GetLargeFont()->DrawAligned(&pPlayerGUIBitmap, drawStartLeft.m_X, drawStartLeft.m_Y, m_Text, m_Alignment);
+				g_FrameMan.GetLargeFont()->DrawAligned(&pPlayerGUIBitmap, drawStartRight.m_X, drawStartRight.m_Y, m_Text, m_Alignment);
+			}
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void PrimitiveMan::BitmapPrimitive::Draw(BITMAP *pDrawScreen, Vector targetPos) {
+		if (!m_pBitmap) {
+			return;
+		}
+		Vector pos[2];
+		unsigned short passes = 0;
```

Comment:
This variable isn't actually used anywhere. It's set in a couple places but not used for anything. In short, delete it.